src/WaxSigningApi.ts

```diff
@@ -274,26 +288,37 @@ export class WaxSigningApi {
       isTemp,
       createData,
       avatar_url: avatarUrl,
-      trustScore
+      trustScore,
+      proof,
     } = event.data;
-
+    let isProofVerified = false;
     if (!verified) {
       throw new Error("User declined to share their user account");
     }
 
     if (!userAccount || !pubKeys) {
       throw new Error("User does not have a blockchain account");
     }
+    if (proof?.verified && this.nonce) {
+      // handle proof logic
+      const message = `cloudwallet-verification-${proof.data.referer}-${this.nonce}-${userAccount}`;
+      isProofVerified = ecc.verify(
```

Comment:
As I said in the `getProofRequiredKeys` function, we likely have to cycle through all keys returned since the account may have been soft claimed which means the wax backend key for creating the signature may not be at index 0

Reply:
@porkchop this logic I adapted from our existed waxProof function which are already worked on prod, could you have a look there as well?

Reply:
right you are correct @danghoangthien my bad!!